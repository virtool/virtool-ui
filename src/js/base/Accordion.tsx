import * as RadixAccordion from "@radix-ui/react-accordion";

/** the base accordion container */
export const Accordion = RadixAccordion.Root;
