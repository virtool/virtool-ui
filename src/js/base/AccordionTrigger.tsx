import { getColor } from "@app/theme";
import * as RadixAccordion from "@radix-ui/react-accordion";
import styled from "styled-components";

/** button for toggling the display of accordion contents  */
const AccordionTrigger = styled(RadixAccordion.Trigger)`
    align-items: center;
    background-color: ${(props) => props.theme.color.white};
    border: none;
    display: flex;
    justify-content: space-between;
    padding: 10px 15px;
    width: 100%;

    &:hover {
        background-color: ${(props) =>
            getColor({ color: "greyHover", theme: props.theme })};
    }
`;

export default AccordionTrigger;
