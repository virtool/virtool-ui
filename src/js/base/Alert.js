import { get } from "lodash-es";
import React from "react";
import styled from "styled-components";
import { getColor, getFontWeight } from "../app/theme";
import { Icon } from "./Icon";

export const getAlertBackgroundColor = ({ color, theme }) =>
    get(theme, ["color", `${color}Lightest`], theme.color.greyLightest);

export const getAlertTextColor = ({ color, theme }) => get(theme, ["color", `${color}Dark`], theme.color.greyDarkest);

const AlertInner = styled.div`
    align-items: ${props => (props.level ? "center" : "normal")};
    border-left: 10px solid ${getColor};
    display: ${props => (props.block ? "block" : "flex")};
    padding: 15px;
`;

const StyledAlert = styled.div`
    background-color: ${getAlertBackgroundColor};
    border-radius: ${props => props.theme.borderRadius.md};
    border: none;
    box-shadow: ${props => props.theme.boxShadow.sm};
    color: ${getAlertTextColor};
    font-weight: ${getFontWeight("thick")};
    margin-bottom: 16px;
    overflow: hidden;

    i.fas:first-child {
        color: currentColor;
        padding-right: 5px;
    }
`;

export const _Alert = ({ block, children, className, color, icon, level }) => (
    <StyledAlert color={color}>
        <AlertInner className={className} block={block} color={color} level={level}>
            {icon ? <Icon name={icon} /> : null}
            {children}
        </AlertInner>
    </StyledAlert>
);

export const Alert = styled(_Alert)``;
