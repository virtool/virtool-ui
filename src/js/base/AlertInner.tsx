import styled, { DefaultTheme } from "styled-components";
import { getColor } from "../app/theme";

type AlertInnerProps = {
    block?: boolean;
    color?: string;
    level?: boolean;
    theme?: DefaultTheme;
};

export const AlertInner = styled.div<AlertInnerProps>`
    align-items: ${props => (props.level ? "center" : "normal")};
    border-left: 10px solid ${getColor};
    display: ${props => (props.block ? "block" : "flex")};
    padding: 15px;
`;

AlertInner.displayName = "AlertInner";
