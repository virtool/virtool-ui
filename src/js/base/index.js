export { Accordion } from "./accordion/Accordion";
export { AccordionContent } from "./accordion/AccordionContent";
export { AccordionTrigger } from "./accordion/AccordionTrigger";
export { ScrollingAccordionItem } from "./accordion/ScrollingAccordionItem";
export { Alert } from "./Alert";
export { AlertInner } from "./AlertInner";
export { AlertOuter } from "./AlertOuter";
export { Attribution } from "./Attribution";
export { AttributionWithName } from "./AttributionWithName";
export { Badge } from "./Badge";
export { Box } from "./Box";
export { BoxGroup } from "./BoxGroup";
export { BoxGroupDisabled } from "./BoxGroupDisabled";
export { BoxGroupHeader } from "./BoxGroupHeader";
export { BoxGroupSearch } from "./BoxGroupSearch";
export { BoxGroupSection } from "./BoxGroupSection";
export { BoxGroupTable } from "./BoxGroupTable";
export { BoxSpaced } from "./BoxSpaced";
export { BoxTitle } from "./BoxTitle";
export { device } from "./Breakpoints";
export { Button } from "./Button";
export { Checkbox } from "./Checkbox";
export { CheckboxLabel } from "./CheckboxLabel";
export { Circle } from "./Circle";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./combobox/ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { DividerHorizontal } from "./DividerHorizontal";
export { DividerVertical } from "./DividerVertical";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputGroup } from "./InputGroup";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLabel } from "./InputLabel";
export { InputLevel } from "./InputLevel";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./pagination/Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./select/Select";
export { SelectButton } from "./select/SelectButton";
export { SelectContent } from "./select/SelectContent";
export { SelectItem } from "./select/SelectItem";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { Toolbar } from "./Toolbar";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
