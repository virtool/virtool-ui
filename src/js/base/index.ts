import InputGroup from "@base/InputGroup.js";
import InputLabel from "@base/InputLabel.js";

export { InputGroup, InputLabel };

export { Link } from "wouter";
export { Checkbox } from "./Checkbox";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./Select";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SelectButton } from "./SelectButton";
export { SelectContent } from "./SelectContent";
export { SelectItem } from "./SelectItem";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
