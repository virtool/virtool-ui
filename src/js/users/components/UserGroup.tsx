import { Checkbox, SelectBoxGroupSection } from "@base";
import React from "react";
import styled from "styled-components";

const StyledUserGroup = styled(SelectBoxGroupSection)`
    text-transform: capitalize;
    user-select: none;
`;

type UserGroupTypes = {
    /** The group unique id */
    id: string | number;
    /** The group name */
    name: string;
    /** Whether the group is selected */
    toggled: boolean;
    /** A callback function to handle the selection of a group */
    onClick: (id: string | number) => void;
};

/**
 * A condensed user group for use in a list of user groups
 */
export function UserGroup({ id, name, toggled, onClick }: UserGroupTypes) {
    return (
        <StyledUserGroup active={toggled} onClick={() => onClick(id)}>
            <Checkbox checked={toggled} label={name} />
        </StyledUserGroup>
    );
}
