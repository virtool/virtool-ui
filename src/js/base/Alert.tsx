import React from "react";
import styled from "styled-components";
import { AlertInner } from "./AlertInner";
import { AlertOuter } from "./AlertOuter";
import { Icon } from "./Icon";

type AlertProps = {
    block?: boolean;
    children: React.ReactNode;
    className?: string;
    color?: string;
    icon?: string;
    level?: boolean;
};

export const Alert = styled(({ block, children, className, color, icon, level }: AlertProps) => (
    <AlertOuter color={color}>
        <AlertInner className={className} block={block} color={color} level={level}>
            {icon ? <Icon name={icon} /> : null}
            {children}
        </AlertInner>
    </AlertOuter>
))``;

Alert.displayName = "Alert";
