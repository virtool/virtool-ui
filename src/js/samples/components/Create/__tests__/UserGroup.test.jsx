import { Select } from "../../../../base";
import { SampleUserGroup } from "../UserGroup";

describe("SampleUserGroup", () => {
    let props;
    beforeEach(() => {
        props = {
            groups: [{ name: "bar", id: "bar_id" }],
            onChange: vi.fn(),
            selected: "bar"
        };
    });

    it("should render", () => {
        const wrapper = shallow(<SampleUserGroup {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should call onChange when InputError is changed", () => {
        const e = {
            target: "foo"
        };
        const wrapper = shallow(<SampleUserGroup {...props} />);
        wrapper.find(Select).simulate("change", e);
        expect(props.onChange).toHaveBeenCalledWith({ target: "foo" });
    });
});
