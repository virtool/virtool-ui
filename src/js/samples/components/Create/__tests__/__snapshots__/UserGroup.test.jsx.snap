// Vitest Snapshot v1

exports[`SampleUserGroup > should render 1`] = `
<InputGroup>
  <styled.label>
    User Group
  </styled.label>
  <Select
    onChange={[MockFunction spy]}
    value="bar"
  >
    <option
      key="none"
      value="none"
    >
      None
    </option>
    <styled.option
      key="bar_id"
      value="bar_id"
    >
      bar
    </styled.option>
  </Select>
</InputGroup>
`;
